esphome/components/light/__init__.py

```diff
@@ -145,6 +145,28 @@
             [cv.percentage], cv.Length(min=3, max=4)
         ),
         cv.Optional(CONF_POWER_SUPPLY): cv.use_id(power_supply.PowerSupply),
+        cv.Optional("power_limits"): cv.Schema(
+            {
+                cv.Required("max_current"): cv.All(cv.current, cv.Range(min=0.0)),
```

Comment:
If changing to `current_limits` as above, `current` suffix is redundant here.
```suggestion
                cv.Required("max"): cv.All(cv.current, cv.Range(min=0.0)),
```